src/Sarif.Converters/TSLintConverter.cs

```diff
@@ -0,0 +1,147 @@
+
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            output.Initialize(id: null, automationId: null);
+
+            TSLintLog tsLintLog = loader.ReadLog(input);
+
+            Tool tool = new Tool
+            {
+                Name = "TSLint"
+            };
+
+            output.WriteTool(tool);
+
+            var results = new List<Result>();
+            foreach(TSLintLogEntry entry in tsLintLog)
+            {
+                results.Add(CreateResult(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+            
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result CreateResult(TSLintLogEntry tslEntry)
+        {
+            tslEntry = tslEntry ?? throw new ArgumentNullException(nameof(tslEntry));
+
+            Result result = new Result()
+            {
+                RuleId = tslEntry.RuleName,
+                Message = tslEntry.Failure
+            };
+
+            switch (tslEntry.RuleSeverity)
+            {
+                case "ERROR":
+                    result.Level = ResultLevel.Error;
+                    break;
+                case "WARN":
+                case "WARNING":
+                    result.Level = ResultLevel.Warning;
+                    break;
+                case "DEFAULT":
+                    result.Level = ResultLevel.Default;
+                    break;
+                default:
+                    result.Level = ResultLevel.NotApplicable;
+                    break;
+            }
+
+            result.Locations = new List<Location>();
+
+            Region region = new Region()
+            {
+                // The TSLint logs have line and column start at 0, Sarif has them starting at 1, so add 1 to each
+                StartLine = tslEntry.StartPosition.Line + 1,
+                StartColumn = tslEntry.StartPosition.Character + 1,
+                EndLine = tslEntry.EndPosition.Line + 1,
+                EndColumn = tslEntry.EndPosition.Character + 1,
+
+                Offset = tslEntry.StartPosition.Position
+            };
+
+            int length = tslEntry.EndPosition.Position - tslEntry.StartPosition.Position + 1;
+            region.Length = length > 0 ? length : 0;
+
+            Uri analysisTargetUri = new Uri(tslEntry.Name, UriKind.Relative);
+
+            PhysicalLocation analysisTarget = new PhysicalLocation(uri: analysisTargetUri, uriBaseId: null, region: region);
+            Location location = new Location();
+            location.AnalysisTarget = analysisTarget;
+
+            result.Locations.Add(location);
```

Comment:
Rather than creating an empty list, creating a Location object, and adding it to the list, you could write:
```
result.Locations = new List<Location> { location };
```

Reply:
I see this is fixed.